### PR TITLE
[ShareWithYou] Add support for Xcode 14 beta 6.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

File: src/sharedwithyou.cs [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+#else
+	[Protocol, Model (AutoGeneratedName = true)]
+#endif
+	[BaseType (typeof (NSObject))]
+	interface SWCollaborationViewDelegate
+	{
+
+		[DesignatedInitializer]
+		[Export ("initWithFrame:")]
+		NativeHandle Constructor (CGRect frame);
```
Comment (dalexsoto) [inline src/sharedwithyou.cs:232]: This needs to be inlined in the implementor classes 
```suggestion
		// [DesignatedInitializer]
		// [Export ("initWithFrame:")]
		// NativeHandle Constructor (CGRect frame);
```

File: tests/introspection/iOS/iOSApiCtorInitTest.cs [hunk 2]
```diff
@@ -255,6 +255,18 @@ protected override bool Skip (Type type)
 			return base.Skip (type);
 		}
 
+		protected override bool SkipCheckShouldReExposeBaseCtor (Type type)
+		{
+			switch(type.Name) {
```
Comment (dalexsoto) [inline tests/introspection/iOS/iOSApiCtorInitTest.cs:260]:
```suggestion
			switch (type.Name) {
```

**Review by dalexsoto — APPROVED** [review]
Other than the two nits, looks good to me.

Protocols should not have init methods on them since we can't really do much from the C# point of view.

File: src/sharedwithyou.cs [hunk 3]
```diff
@@ -0,0 +1,399 @@
...
+using CoreFoundation;
+using CoreGraphics;
+using Foundation;
+using SharedWithYouCore; 
+using UniformTypeIdentifiers;
+using ObjCRuntime;
+
+#if MONOMAC
+using AppKit;
+using UIViewController=System.Object;
```
Comment (rolfbjarne) [inline src/sharedwithyou.cs:22]: Why not:

```suggestion
using UIViewController=AppKit.NSViewController;
```

**Review by rolfbjarne — APPROVED** [review]

**Review by chamons — APPROVED** [review]

File: src/sharedwithyou.cs [hunk 4]
```diff
@@ -0,0 +1,399 @@
...
+
+		[Export ("contentType", ArgumentSemantic.Copy)]
+		UTType ContentType { get; }
+	}
+
+	interface ISWHighlightEvent {}
+
+	[NoWatch, NoTV, NoMac, iOS (16,0), MacCatalyst (16,0)]
+	[Protocol]
+	[BaseType (typeof (NSObject))]
```
Comment (mandel-macaque) [inline src/sharedwithyou.cs:335]:
```suggestion
```

File: src/sharedwithyou.cs [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+	[Protocol, Model]
+#else
+	[Protocol, Model (AutoGeneratedName = true)]
+#endif
+	[BaseType (typeof (NSObject))]
+	interface SWHighlightCenterDelegate
+	{
+		[Abstract]
+		[Export ("highlightCenterHighlightsDidChange:")]
+		void HighlightCenterHighlightsDidChange (SWHighlightCenter highlightCenter);
```
Comment (tj-devel709) [inline src/sharedwithyou.cs:162]:
```suggestion
		void DidHighlightCenterHighlightsChange (SWHighlightCenter highlightCenter);
```
perhaps verbify - not sure if this changes a pattern

File: src/sharedwithyou.cs [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+		[Export ("collaborationHighlightForIdentifier:error:")]
+		[return: NullAllowed]
+		SWCollaborationHighlight GetCollaborationHighlight (string collaborationIdentifier, [NullAllowed] out NSError error);
+
+		[Async]
+		[Export ("getCollaborationHighlightForURL:completionHandler:")]
+		void GetCollaborationHighlight (NSUrl url, Action<SWCollaborationHighlight, NSError> completionHandler);
+
+		[Export ("postNoticeForHighlightEvent:")]
+		void PostNoticeForHighlightEvent (ISWHighlightEvent @event);
```
Comment (tj-devel709) [inline src/sharedwithyou.cs:200]:
```suggestion
		void PostNotice (ISWHighlightEvent @event);
```
perhaps

**Review by tj-devel709 — APPROVED** [review]
Few Suggestions but LGTM

File: src/sharedwithyou.cs (hunk 5 above)
Comment (mandel-macaque) [reply src/sharedwithyou.cs:162]: I think it is better to do
```suggestion
		void HighlightsDidChange (SWHighlightCenter highlightCenter);
```

**Pull request merged**